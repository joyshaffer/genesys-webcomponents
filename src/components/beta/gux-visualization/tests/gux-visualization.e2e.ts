import { newE2EPage } from '@stencil/core/testing';

const visualizationSpecLine = {
  $schema: 'https://vega.github.io/schema/vega-lite/v5.json',
  data: { url: 'https://vega.github.io/editor/data/seattle-weather.csv' },
  mark: {
    type: 'line',
    interpolate: 'monotone'
  },
  encoding: {
    x: { timeUnit: 'month', field: 'date', bandPosition: 0.5 },
    y: { aggregate: 'mean', field: 'temp_max' }
  }
};

describe('gux-visualization-beta', () => {
  describe('#render', () => {
    [
      {
        description: 'should render default button',
        html: '<gux-visualization-beta lang="en" id="visualization-1"></gux-visualization-beta>'
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newE2EPage({ html });
        const element = (await page.find(
          'gux-visualization-beta'
        )) as HTMLGuxVisualizationElement;
        element.visualizationSpec = visualizationSpecLine;

        await page.waitForChanges();

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
