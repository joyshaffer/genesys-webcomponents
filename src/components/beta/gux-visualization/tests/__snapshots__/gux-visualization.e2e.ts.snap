// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-visualization-beta #render should render default button 1`] = `"<gux-visualization-beta lang=\\"en\\" id=\\"visualization-1\\" class=\\"vega-embed hydrated\\" role=\\"graphics-document\\" aria-roledescription=\\"visualization\\" aria-label=\\"Vega visualization\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" class=\\"marks\\" width=\\"0\\" height=\\"0\\" viewBox=\\"0 0 0 0\\"><g fill=\\"none\\" stroke-miterlimit=\\"10\\" transform=\\"translate(0,0)\\"><g class=\\"mark-group role-frame root\\" role=\\"graphics-object\\" aria-roledescription=\\"group mark container\\"><g transform=\\"translate(0,0)\\"><path class=\\"background\\" aria-hidden=\\"true\\" d=\\"M0,0h0v0h0Z\\"></path><g></g><path class=\\"foreground\\" aria-hidden=\\"true\\" d=\\"\\" display=\\"none\\"></path></g></g></g></svg><form class=\\"vega-bindings\\"></form></gux-visualization-beta>"`;
